components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/KeyManagersApiServiceImpl.java

```diff
@@ -156,8 +167,30 @@ public Response keyManagersPost(KeyManagerDTO body, MessageContext messageContex
             return Response.created(location)
                     .entity(KeyManagerMappingUtil.toKeyManagerDTO(createdKeyManagerConfiguration)).build();
         } catch (URISyntaxException e) {
-            String error = "Error while Creating Key Manager configuration in organization " + organization;
+            String error = "Error while creating Key Manager configuration in organization " + organization;
             throw new APIManagementException(error, e, ExceptionCodes.INTERNAL_ERROR);
+        } catch (IllegalArgumentException e) {
+            String error = "Error while storing Key Manager permission roles with name "
+                    + body.getName() + " in tenant " + organization;
+            throw new APIManagementException(error, e, ExceptionCodes.ROLE_DOES_NOT_EXIST);
+        }
+    }
+
+    public void validatePermissions (KeyManagerPermissionConfigurationDTO permissionDTO)
+            throws IllegalArgumentException, APIManagementException{
```

Comment:
Formatting

Reply:
Fixed in 1d58204afd0d2b23f33e003063649b697cf2e32a